src/include/migraphx/shape.hpp

```diff
@@ -132,11 +134,11 @@ struct shape
 
     shape(type_t t, std::vector<dynamic_dimension> dims);
 
-    // Construct a dynamic shape from three sets of lengths (of the same rank)
+    // Construct a dynamic shape from vectors of mins, maxes, and opts
```

Comment:
Add a docstring why `opts_list` has a set of optimal values for each dynamic-dimension.